### PR TITLE
feat(init): initial setup

### PR DESCRIPTION
**Pull request by qingyashu**
Now as a start, an empty npm module is setup published at: https://www.npmjs.com/package/@gen3/base-component

Now user can import module by (fake button example): 
```
import Button from '@gen3/base-component/dist/components/button/Button';
<Button label="hello-world" />
```

We have a lot to discuss, now I only think of following questions:
1. About source file structure, for now is empty, do we have any convention or preference 
2. About the way we distribute components. For now it's using `npm prepare` to let babel precompile into "dist" directories, do we have better idea? 
3. About UI demo page: we can use storybook for now. But as we are deploying on github, we can have a githubpages with more beautiful layout (I have no idea for now...maybe some fancy blog template like hugo or jekyll?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+language: node_js
+git:
+  depth: 1
+node_js:
+  - "8"
```
Comment (abgeorge7) [inline .travis.yml]: I think we want this to be 10 now - @frickjack ?

File: src/components/button/Button.test.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import Button from './Button';
+
+describe('<Button />', () => {
+  const btn = mount(<Button label="tes-button"></Button>).find(Button);
```
Comment (abgeorge7) [inline src/components/button/Button.test.js]: This is picky but I think typo - "test-button" 😬

**Review by abgeorge7 — COMMENTED** [review]
Two quick fixes

File: src/components/button/Button.test.js (hunk 2 above)
Comment (abgeorge7) [reply src/components/button/Button.test.js]: If this is just a temporary file to be used a placeholder, no worries on fixing this

Comment (qingyashu) [reply src/components/button/Button.test.js]: It's temporary, but I will fix this anyway 😛 Thanks!!

**Comment by abgeorge7** [comment]
Thanh brought up that if we want to make this reusable it would be easier for users if we just had .css files vs. .less files - thoughts?

**Comment by qingyashu** [comment]
@abgeorge7  @thanh-nguyen-dang  Can you explain reason why using .css? I tried let data-portal import the new package and the style works fine.

**Comment by thanh-nguyen-dang** [comment]
`.less` needs a special pre-processor while `.css` is a standard one. @qingyashu

**Comment by qingyashu** [comment]
You are correct!
Is there any way to pre-compile less into .css? I think the reason we would like to use  .less is to take advantage of defining variables. You have any better idea? @frickjack @thanh-nguyen-dang

File: src/css/base.css [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+}
+.form-special-number {
+  font-size: 18px; 
+  font-weight: 600; 
+  line-height: 1em; 
+  letter-spacing: .02rem; 
+  color: #3283C8;
+}
+
+.gen3-color-base-blue {
```
Comment (abgeorge7) [inline src/css/base.css]: is the purpose of these color classes to replace our `.less` variables?

File: stories/button.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import '../src/css/base.css'
+
+storiesOf('Components/Buttons', module)
+  .add('Button Primary White', () => (
+    <button className="btn-primary-white">btn-primay-white</button>
```
Comment (abgeorge7) [inline stories/button.js]: typo in the button label 😬 `btn-primary-white`

File: stories/button.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  ))
+  .add('Button Primary Orange', () => (
+    <button className="btn-primary-orange">btn-primay-white</button>
```
Comment (abgeorge7) [inline stories/button.js]: I think this label should be `btn-primary-orange`

**Comment by abgeorge7** [comment]
Oh also for the Data Visualization Colors story, there's a warning about the children needing `key`s

File: src/css/base.less [hunk 6]
```diff
@@ -0,0 +1,344 @@
+* {
```
Comment (abgeorge7) [inline src/css/base.less:1]: do we still need this file or are we getting rid of it?

**Review by abgeorge7 — COMMENTED** [review]
I'm still kind of confused on how we're handling variables (for things like color and font). Right now we have a CSS and Less file, but don't we only want one? I think we can do CSS variables with something like this:
https://www.w3schools.com/css/css3_variables.asp

**Comment by qingyashu** [comment]
@abgeorge7 @thanh-nguyen-dang  Could you review this PR if you have time? For now I have moved some color variables, some icons, and button component into this repo. I think for now we can still keep `base.less` for `data-portal` repo to import, but we are adding new styles into `base.css` step by step. Any comments about styles/variable/icons/components welcome!

File: dist/components/Button/Button.test.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+'use strict';
```
Comment (abgeorge7) [inline dist/components/Button/Button.test.js]: I think we can exclude tests from the `dist` folder right? users won't need them

File: src/components/Button/index.jsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    if (this.props.onClick) {
+      this.props.onClick(e);
+    }
+  }
+
+  render() {
+    return (
+      <button
+        type="button"
+        className={`${this.props.className} g3-button g3-button--${this.props.type} g3-button--${this.props.enabled ? 'enabled' : 'disabled'}`}
```
Comment (abgeorge7) [inline src/components/Button/index.jsx]: are there any classes with `'enabled'` in them? I didn't see any

File: src/components/Button/index.jsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      >
+        {this.props.label}
+      </button>
+    );
+  }
+}
+
+Button.propTypes = {
+  label: PropTypes.string.isRequired,
+  type: PropTypes.oneOf(['primary', 'secondary']),
```
Comment (abgeorge7) [inline src/components/Button/index.jsx]: I think `type` should be renamed to `buttonType` or something

File: src/css/icon.css [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+  background-size: 18px 18px;
+}
+
+.g3-icon--sm {
+  width: 12px;
+  height: 12px;
+  background-size: 12px 12px;
+}
+
+.g3-icon--analysis {
```
Comment (abgeorge7) [inline src/css/icon.css]: so the idea with icons is to use them as div background images? I'm wondering if we should rethink how we do icons in general

File: src/css/icon.css [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+.g3-icon--profile {
+  background-image: url(../images/icons/profile.svg);
+}
+
+.g3-icon--query {
+  background-image: url(../images/icons/query.svg);
+}
+
+.g3-icon--upload-white {
```
Comment (abgeorge7) [inline src/css/icon.css]: this is a good example of why I think we should rethink icons - we should have a separate image for a color change. Maybe we can talk about it and make a new ticket?

File: stories/typography.js [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+  borderBottom: 'solid gray 1px',
+  paddingTop: '10px',
+  paddingLeft: '10px',
+};
+
+storiesOf('General/Typography', module)
+  .add('Typography normal', () => (
+    <div>
+      <div className="typo-demo-block" style={typoDemoItemStyle}>
+        <h1 className="h1">
```
Comment (abgeorge7) [inline stories/typography.js]: I don't think we need a `h1` class, in our `base.css` we should just override the h1 element itself
Same with h2, h3, etc.

**Review by abgeorge7 — CHANGES_REQUESTED** [review]
Nice! love the variables. I had a few comments, and I also noticed some committed `DS_Store` files.

I think it'd be good to make some tickets for improvements/future PRs, or even add `TODO`s in the code for things like icons, explaining when to use which font, which icon, etc.

File: src/css/icon.css (hunk 10 above)
Comment (qingyashu) [reply src/css/icon.css]: Yes! Let's have more discussion about icons! 💪

File: src/css/icon.css (hunk 11 above)
Comment (qingyashu) [reply src/css/icon.css]: Ticket PXD-1696 created

**Review by abgeorge7 — APPROVED** [review]

**Pull request merged**